pset-iterators.js

```diff
@@ -184,6 +220,21 @@ console.log('Problem 7:')
 
 // Add your code below this line
 
+//  grab a list of presidents who have been in office for less than 4 years
+const shortTermPresidents = presidents.filter(presidentObj => {
+  if (presidentObj.left_office === null) {
+    //  if the president never left office, calculate based off the cufrent year (2019)
+    return (2019 - presidentObj.took_office.split('-')[0]) < 4
+  } else {
+    //  below line calculates the year the president took office minus
+    //  the year the president left office and retuns the object if the Result
+    //  is less than 4
+    return (presidentObj.left_office.split('-')[0] - presidentObj.took_office.split('-')[0]) < 4
+  }
+})
+
+//  print to console
+console.log(shortTermPresidents)
```

Comment:
@sad3xoxo  Good work here on this problem using the filter method. Good job utilizing the conditional to determine the outcomes for the current president and past presidents

Take a look at this approach [here](https://github.com/jsr20190826/hw-03-iterators-solution/blob/96bab20deba3ad116374a5827291a951faf5d555/pset-iterators.js#L201)